code/modules/surgery/bodyparts/helpers.dm

```diff
@@ -15,6 +15,7 @@
 /mob/living/carbon/proc/del_and_replace_bodypart(obj/item/bodypart/new_limb, special)
 	var/obj/item/bodypart/old_limb = get_bodypart(new_limb.body_zone)
 	if(old_limb)
+		old_limb.drop_limb(special = TRUE)
 		qdel(old_limb)
```

Comment:
this shouldn't be necessarily, qdeling a limb should implicitly specially remove it

Reply:
I blackmailed melon into merging it. I’ll look into it when I get to my computer, I forgot the finer details of drop_limb